pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,185 @@
+	secondaryNetworks := vmispec.FilterNetworksSpec(networks, func(network virtv1.Network) bool {
+		return vmispec.IsSecondaryMultusNetwork(network)
+	})
+
+	networkToIPAMClaimName := make(map[string]string)
+	for _, network := range secondaryNetworks {
+		if allowPersistentIPs, err := getAllowPersistentIPs(virtClient, namespace, network.Multus.NetworkName); err != nil {
+			return map[string]string{}, fmt.Errorf("failed retrieving allowPersistentIPs flag: %w", err)
+		} else if allowPersistentIPs {
+			networkToIPAMClaimName[network.Name] = fmt.Sprintf("%s.%s", vmiName, network.Name)
+		}
+	}
+
+	return networkToIPAMClaimName, nil
+}
+
+func getAllowPersistentIPs(virtClient kubecli.KubevirtClient, namespace string, fullNetworkName string) (bool, error) {
+	namespace, networkName := getNamespaceAndNetworkName(namespace, fullNetworkName)
+	nad, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Get(context.Background(), networkName, v1.GetOptions{})
+	if err != nil {
+		return false, fmt.Errorf("failed to locate network attachment definition %s/%s", namespace, networkName)
+	}
+
+	if nad.Spec.Config == "" {
+		return false, nil
+	}
+
+	netConf := struct {
+		AllowPersistentIPs bool `json:"allowPersistentIPs,omitempty"`
+	}{}
+	err = json.Unmarshal([]byte(nad.Spec.Config), &netConf)
+	if err != nil {
+		return false, fmt.Errorf("failed to unmarshal NAD spec.config JSON: %v", err)
+	}
+
+	return netConf.AllowPersistentIPs, nil
+}
+
+func CreateIPAMClaims(client kubecli.KubevirtClient, vmi *virtv1.VirtualMachineInstance, ownerRef *v1.OwnerReference) error {
+	nonAbsentNetworks := filterNonAbsentNetworks(vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks)
+	networkToIPAMClaimName, err := GetNetworkToIPAMClaimName(client, vmi.Namespace, vmi.Name, nonAbsentNetworks)
+	if err != nil {
+		return fmt.Errorf("failed composing networkToIPAMClaimName: %w", err)
+	}
+	persistentIPnetworks := vmispec.FilterNetworksSpec(nonAbsentNetworks, func(network virtv1.Network) bool {
+		return networkToIPAMClaimName[network.Name] != ""
+	})
+
+	// For a standalone VMI, the IPAMClaim owner is the VMI itself
+	if ownerRef == nil {
+		ownerRef = v1.NewControllerRef(vmi, virtv1.VirtualMachineInstanceGroupVersionKind)
+	}
+
+	claims := composeIPAMClaims(vmi.Namespace, ownerRef, persistentIPnetworks, networkToIPAMClaimName)
+	err = createIPAMClaims(client, vmi.Namespace, claims)
+	if err != nil {
+		return fmt.Errorf("failed IPAMClaims creation for VMI %s: %w", vmi.Name, err)
+	}
+
+	return nil
+}
+
+func composeIPAMClaims(namespace string, ownerRef *v1.OwnerReference, persistentIPnetworks []virtv1.Network, networkToIPAMClaimName map[string]string) []*ipamclaims.IPAMClaim {
+	claims := []*ipamclaims.IPAMClaim{}
+	networkNameScheme := namescheme.CreateHashedNetworkNameScheme(persistentIPnetworks)
+	for _, network := range persistentIPnetworks {
```

Comment:
You have to iterate networkToIPAMClaimName instead of persistentIPnetwork, so you can remove that argument to, at the end you need the network name and that's already the keys at `networkToIPAMClaimName`

Reply:
we need `network.Multus.NetworkName` which we don't have in the map 
map has only network.Name as key, and the ipam-claim-reference as value